bvm/ballerina-runtime/README.md

```diff
@@ -0,0 +1,248 @@
+| io.ballerina.runtime.api.Runtime         | An instance of the current runtime can be obtained through an Environment instance. This will contain APIs to call Ballerina object methods asynchronously.                                                                                                                                                                                                                         |
+| io.ballerina.runtime.api.TypeTags        | Contains runtime type tags.                                                                                                                                                                                                                                                                                                                                                 |
+
+## Create a Ballerina value
+
+The **io.ballerina.runtime.api.creators.ValueCreator**  provides APIs to create Ballerina value instances. A Ballerina value can be created using one of the methods provided in the ValueCreator class as per the requirements.
+For example, an array value can be created as follows,
+
+```
+BArray array = ValueCreator.createArrayValue(arrayType);
+```
+
+## Create a Ballerina type
+
+`io.ballerina.runtime.api.creators.TypeCreator` class provides APIs to create Ballerina type instances. There are multiple methods available to create Ballerina type instances based on the requirements.
+
+For example, a string array type can be created as follows,
+
+```
+ArrayType strArrayType = TypeCreator.createArrayType(PredefinedTypes.TYPE\_STRING)
+```
+
+## Calling Ballerina object method
+
+Ballerina runtime exposes APIs to call the Ballerina object method and Ballerina function pointer using Java.
+
+`io.ballerina.runtime.api.Runtime` class exposes two Java APIs to call object methods.
+
+1. invokeMethodAsyncSequentially
+
+Invoke the Object method asynchronously and sequentially. This method will ensure that the object methods are invoked in the same thread where other object methods are executed. So, the methods will be executed sequentially per object level.
+
+2. invokeMethodAsyncConcurrently
+
+Invoke the Object method asynchronously and concurrently. The caller needs to ensure that no data race is possible for the mutable state with a given object method and with arguments. So, the method can be concurrently run with different OS threads.
+
+**Note:** If the caller can ensure that the given object and object method are isolated and no data race is possible for the mutable state with given arguments, use @invokeMethodAsyncConcurrently
+otherwise, use @invokeMethodAsyncSequentially. We can decide the object method isolation if and only if both object.getType().isIsolated() and object.getType().isIsolated(methodName) returns true.
```

Comment:
```suggestion
>**Note:** If the caller can ensure that the given object and object method are isolated and no data race is possible for the mutable state with given arguments, use `invokeMethodAsyncConcurrently`, otherwise use `invokeMethodAsyncSequentially`. We can decide the object method isolation if and only if both `object.getType().isIsolated()` and `object.getType().isIsolated(methodName)` return `true`.
```